src/KingSystem/Damage/dmgDamageManagerBase.cpp

```diff
@@ -0,0 +1,271 @@
+#include "dmgDamageManagerBase.h"
+#include "Game/DLC/aoc2.h"
+#include "KingSystem/ActorSystem/actActor.h"
+#include "KingSystem/ActorSystem/actActorConstDataAccess.h"
+#include "KingSystem/ActorSystem/actActorParamMgr.h"
+#include "KingSystem/ActorSystem/actLifeRecoveryInfo.h"
+#include "KingSystem/Resource/GeneralParamList/resGParamListObjectGeneral.h"
+#include "KingSystem/Resource/resResourceGParamList.h"
+
+namespace ksys::dmg {
+
+DamageManagerBase_UnknownBase1::DamageManagerBase_UnknownBase1(ksys::act::Actor* WeaponActor)
+    : mActor(WeaponActor) {}
+
+// Compiler seems to combine zero(0) writes to (0x0 ,0x8) and (0x10, 0x18)
+// when writing the vtable and Actor.
+// The original Compiler writes (0x8, 0x10) in one 'stp', and writes 0x0 and 0x18 individually with
+// 'str'. The rest seems to fall out of sync due to that, but it's otherwise functionally the same.
+// NON_MATCHING: Incorrect order.
+DamageManagerBase::DamageManagerBase(ksys::act::Actor* WeaponActor)
+    : DamageManagerBase_UnknownBase1(WeaponActor) {}
+
+u32 DamageManagerBase::getDamage() {
+    u32 result;
+    if (canTakeDamage())
+        result = mDamage;
+    else
+        result = 0LL;
+
+    return result;
+}
+
+void DamageManagerBase::addDamageCallback(s32 eventId, DamageCallback* callback) {
+    if (mCallbacks.isBufferReady() && !callback->mDamageManager) {
+        DamageCallback* next = mCallbacks[eventId];
+        if (next) {
+            DamageCallback* prev;
+            while (next) {
+                prev = next;
+                next = next->mNext;
+            };
+
+            prev->mNext = callback;
+            callback->mPrev = prev;
+        } else {
+            mCallbacks[eventId] = callback;
+        }
+
+        callback->mDamageManager = this;
+        callback->mEventId = eventId;
+    }
+}
+
+void DamageManagerBase::removeDamageCallback(DamageCallback* callback) {
+    if (!mCallbacks.isBufferReady() || callback->mDamageManager != this) {
+        return;
+    }
+
+    u32 eventId = callback->mEventId;
+    DamageCallback* currentCallback = mCallbacks[eventId];
```

Comment:
```suggestion
    DamageCallback* current_callback = mCallbacks[eventId];
```
(non-member variables use snake case)